scripts/upgrade-hooks/1.1.0.ts

```diff
@@ -0,0 +1,251 @@
+}
+
+interface Token {
+  id: string;
+  status: number;
+  priceLog: { amount: string | null };
+}
+
+interface Offer {
+  verifierFee: string;
+  facilitatorFeePercent: string;
+  fNFTs: Token[];
+}
+
+export interface FeeData {
+  tokenId: string;
+  bosonProtocolFee: bigint;
+  fermionFeeAmount: bigint;
+  verifierFee: bigint;
+  facilitatorFeeAmount: bigint;
+}
+
+const OFFER_QUERY = `
+  query {
+    offers(where: { fNFTs_: { status_in: [3, 4, 5, 6, 7] } }) {
+      verifierFee
+      facilitatorFeePercent
+      fNFTs {
+        id
+        status
+        priceLog {
+          amount
+        }
+      }
+    }
+  }
+`;
+
+const FNFT_RANGE_QUERY = `
+  query {
+    fnftranges {
+      bosonOfferId
+      startingId
+      quantity
+    }
+  }
+`;
+
+interface FNFTRange {
+  bosonOfferId: string;
+  startingId: string;
+  quantity: string;
+}
+
+export interface OfferData {
+  offerId: string;
+  itemQuantity: string;
+  firstTokenId: string;
+}
+
+/**
+ * Fetch GraphQL data for FNFT ranges.
+ */
+async function fetchFNFTRangeData(client: any): Promise<FNFTRange[]> {
+  const result = await client.query(FNFT_RANGE_QUERY, {}).toPromise();
+
+  if (result.error) {
+    throw new Error(`GraphQL error: ${result.error.message}`);
+  }
+
+  if (!result.data?.fnftranges) {
+    throw new Error("No FNFT range data found in GraphQL response");
+  }
+
+  return result.data.fnftranges;
+}
+
+/**
+ * Prepare backfill data for offers.
+ */
+export async function prepareOfferBackfillData(client: any): Promise<OfferData[]> {
+  const fnftRanges = await fetchFNFTRangeData(client);
+  const offerDataList: OfferData[] = [];
+
+  for (const range of fnftRanges) {
+    offerDataList.push({
+      offerId: range.bosonOfferId,
+      itemQuantity: range.quantity,
+      firstTokenId: range.startingId,
+    });
+  }
+
+  console.log("Prepared Offer Backfill Data:", JSON.stringify(offerDataList, null, 2));
+
+  return offerDataList;
+}
+
+/**
+ * Fetch GraphQL data for offers and tokens in specific states.
+ */
+async function fetchGraphQLData(client: any): Promise<Offer[]> {
+  const result = await client.query(OFFER_QUERY, {}).toPromise();
+
+  if (result.error) {
+    throw new Error(`GraphQL error: ${result.error.message}`);
+  }
+
+  if (!result.data?.offers) {
+    throw new Error("No data found in GraphQL response");
+  }
+
+  return result.data.offers;
+}
+
+/**
+ * Calculate fees for backfilling based on offer details.
+ */
+function calculateFees(verifierFee: bigint, facilitatorFeePercentBps: bigint, tokenId: string, price: bigint): FeeData {
+  const HUNDRED_PERCENT_BPS = 100_00n;
+  const BOSON_PROTOCOL_FEE_BPS = 50n; // 0.5%
+  const bosonProtocolFee = (price * BOSON_PROTOCOL_FEE_BPS) / HUNDRED_PERCENT_BPS;
+  const fermionFeeAmount = 0n;
+  const remainder = price - bosonProtocolFee - fermionFeeAmount - verifierFee;
+  const facilitatorFeeAmount = remainder > 0n ? (remainder * facilitatorFeePercentBps) / HUNDRED_PERCENT_BPS : 0n;
```

Comment:
```suggestion
  const facilitatorFeeAmount =  (remainder * facilitatorFeePercentBps) / HUNDRED_PERCENT_BPS;
```
Should be the same without the ternary operator. If `remainder` is negative, something went wrong before that.